library/core/src/panic/location.rs

```diff
@@ -179,24 +208,34 @@ impl<'a> Location<'a> {
     pub const fn column(&self) -> u32 {
         self.col
     }
-}
 
-#[unstable(
-    feature = "panic_internals",
-    reason = "internal details of the implementation of the `panic!` and related macros",
-    issue = "none"
-)]
-impl<'a> Location<'a> {
-    #[doc(hidden)]
-    pub const fn internal_constructor(file: &'a str, line: u32, col: u32) -> Self {
-        Location { file, line, col }
+    /// Returns the name of the source file from which the panic originated as a nul-terminated
+    /// string.
+    ///
+    /// This function is like [`Location::file`], except that it returns a nul-terminated string. It
+    /// is mainly useful for passing the filename into C or C++ code.
+    #[must_use]
+    #[inline]
+    #[unstable(feature = "panic_file_with_nul", issue = "none")]
+    #[cfg(not(bootstrap))]
+    pub fn file_with_nul(&self) -> &CStr {
+        let file_with_nul = self.file_with_nul.as_bytes();
+
+        #[cfg(debug_assertions)]
+        if !matches!(file_with_nul.last(), Some(0)) {
```

Comment:
You can remove this, the unchecked constructor does this already.

Reply:
Done.